### ISSUE TITLE
HBoxContainer default separation not shown

### ISSUE THREAD

**Issue by ScepticDope** [godotengine/godot#27947]
**Godot version:**
3.1.1 Mono 2fbc421

**Issue description:**
`HBoxContainer` has a default separation of some kind. But the field for it in the editor, when unchecked, displays 0. This is very confusing. Either it should display the correct separation value or the default should be set to 0.

I would prefer the latter, I was looking for a while for some margin or padding setting on some children nodes that were set to fill and expand because a random gap was between them for no obvious reason and the widths where a bit off.

**Comment by TheDuriel** [comment]
This is true for all Custom Constants for Control nodes. They do not display the default value used because until you click the checkbox they are not actually in use.

It would probably be a good idea to turn most of the custom constants into proper properties, but i imagine the tech behind Control makes this wasteful/difficult.

Then again, you are meant to set these values via themes, and only dig into them for nodes that require unique configurations.

(And can be please merge H and V box containers into one? The difference is literally just flip a bool. But that is another matter.)

**Comment by ScepticDope** [comment]
I like that idea of merging those containers, would give me a neater node tree for different split screen setups. But perhaps that could simply be a new container `VHBoxContainer`?

**Comment by TheDuriel** [comment]
BoxContainer already exists, which V and H box extend.

**Comment by ScepticDope** [comment]
Yes but you can't add it to your scene as a node as of now. Perhaps a new node could be introduced `HVBoxContainer` and that one is switchable to H or V mode from code and in the editor.

**Comment by MaaaxiKing** [comment]
I think, this should have the bug Label!

**Comment by Calinou** [comment]
@MaaaxiKing This is not technically a bug, although the default behavior could be better documented.
